packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_loan_payback.test.ts

```diff
@@ -0,0 +1,760 @@
+  await alice.generate(1)
+
+  // setLoanToken AMZN
+  await alice.rpc.loan.setLoanToken({
+    symbol: 'AMZN',
+    fixedIntervalPriceId: 'AMZN/USD'
+  })
+  await alice.generate(1)
+
+  // mint loan token AMZN
+  await alice.token.mint({ symbol: 'AMZN', amount: 40000 })
+  await alice.generate(1)
+
+  await alice.rpc.account.sendTokensToAddress({}, { [aliceColAddr]: ['40000@AMZN'] })
+  await alice.generate(1)
+
+  // setLoanToken UBER
+  await alice.rpc.loan.setLoanToken({
+    symbol: 'UBER',
+    fixedIntervalPriceId: 'UBER/USD'
+  })
+  await alice.generate(1)
+
+  // mint loan token UBER
+  await alice.token.mint({ symbol: 'UBER', amount: 40000 })
+  await alice.generate(1)
+
+  // createLoanScheme 'scheme'
+  await alice.rpc.loan.createLoanScheme({
+    minColRatio: 500,
+    interestRate: new BigNumber(3),
+    id: 'scheme'
+  })
+  await alice.generate(1)
+  await tGroup.waitForSync()
+
+  bobColAddr = await bProviders.getAddress()
+  await bob.token.dfi({ address: bobColAddr, amount: 30000 })
+  await bob.generate(1)
+  await tGroup.waitForSync()
+
+  // await alice.rpc.account.sendTokensToAddress({}, { [bobColAddr]: ['40@TSLA'] })
+
+  await alice.rpc.account.accountToAccount(aliceColAddr, { [bobColAddr]: '1@BTC' })
+  await alice.generate(1)
+  await tGroup.waitForSync()
+
+  // createVault
+  bobVaultAddr = await bob.generateAddress()
+  bobVaultId = await bob.rpc.loan.createVault({
+    ownerAddress: bobVaultAddr,
+    loanSchemeId: 'scheme'
+  })
+  await bob.generate(1)
+
+  // depositToVault DFI 1000
+  await bob.rpc.loan.depositToVault({
+    vaultId: bobVaultId, from: bobColAddr, amount: '10000@DFI'
+  })
+  await bob.generate(1)
+
+  // depositToVault BTC 1
+  await bob.rpc.loan.depositToVault({
+    vaultId: bobVaultId, from: bobColAddr, amount: '1@BTC'
+  })
+  await bob.generate(1)
+
+  // createVault #2
+  bobVaultAddr1 = await bob.generateAddress()
+  bobVaultId1 = await bob.rpc.loan.createVault({
+    ownerAddress: bobVaultAddr1,
+    loanSchemeId: 'scheme'
+  })
+  await bob.generate(1)
+
+  await bob.rpc.loan.depositToVault({
+    vaultId: bobVaultId1, from: bobColAddr, amount: '10000@DFI'
+  })
+  await bob.generate(1)
+
+  // createVault for liquidation
+  const bobLiqVaultAddr = await bob.generateAddress()
+  bobLiqVaultId = await bob.rpc.loan.createVault({
+    ownerAddress: bobLiqVaultAddr,
+    loanSchemeId: 'scheme'
+  })
+  await bob.generate(1)
+
+  await bob.rpc.loan.depositToVault({
+    vaultId: bobLiqVaultId, from: bobColAddr, amount: '10000@DFI'
+  })
+  await bob.generate(1)
+
+  await bob.rpc.loan.takeLoan({
+    vaultId: bobLiqVaultId,
+    amounts: '100@UBER'
+  })
+  await bob.generate(1)
+  await tGroup.waitForSync()
+
+  // liquidated: true
+  await alice.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '100000@UBER', currency: 'USD' }] })
+  await alice.generate(1)
+  await tGroup.waitForSync()
```

Comment:
I think you need `waitForSync` when you are calling `bob` next. 
You may remove `await tGroup.waitForSync()`